format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/PathTrackingDelimitedInputFormat.java

```diff
@@ -69,7 +69,48 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
         if (delegate.nextKeyValue()) {
           // skip to next if the current record is header
           if (skipHeader && delegate.getCurrentKey().get() == 0L) {
-            return delegate.nextKeyValue();
+            delegate.nextKeyValue();
+          }
+          // getCurrentValue logic moved to nextKeyValue to avoid discrepancies in the data if there are multiple
+          // calls to getCurrentValue method.
+          String delimitedString = delegate.getCurrentValue().toString();
+          builder = StructuredRecord.builder(schema);
+          Iterator<Schema.Field> fields = schema.getFields().iterator();
+          Iterator<String> splitsIterator = getSplitsIterator(enableQuotesValue, delimitedString, delimiter);
+
+          while (splitsIterator.hasNext()) {
+            String part = splitsIterator.next();
+            if (!fields.hasNext()) {
```

Comment:
this is a comment on existing code, but it would be good to move everything in this `if` block into some other method like `handleBadLine()` or something similar. That would make the logic here easier to follow